app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/NoCollectionsMessageViewHolder.kt

```diff
@@ -5,59 +5,210 @@
+) {
+    val normalTabsState = store.observeAsComposableState { state -> state.normalTabs }
+
+    Box(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .dashedBorder(
+                color = FirefoxTheme.colors.borderPrimary,
+                cornerRadius = 8.dp,
+                dashHeight = 2.dp,
+                dashWidth = 4.dp
             )
-            setOnClickListener {
-                interactor.onRemoveCollectionsPlaceholder()
-            }
-        }
+            .padding(
+                start = 16.dp,
+                bottom = 16.dp
+            )
+    ) {
+        Column(
+            Modifier
+                .padding(all = 16.dp)
+                .fillMaxWidth()
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
+            ) {
+                SectionHeader(
+                    text = stringResource(R.string.collections_header),
+                    modifier = Modifier
+                        .weight(1f)
+                        .wrapContentHeight(align = Alignment.Top)
+                )
 
-        store.flowScoped(viewLifecycleOwner) { flow ->
-            flow.map { state -> state.normalTabs.size }
-                .ifChanged()
-                .collect { tabs ->
-                    binding.addTabsToCollectionsButton.isVisible = tabs > 0
+                IconButton(
+                    onClick = { onRemoveCollectionsPlaceholder() }
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_close),
+                        contentDescription = stringResource(
+                            id = R.string.remove_home_collection_placeholder_content_description
+                        ),
+                        modifier = Modifier.size(24.dp),
+                        tint = FirefoxTheme.colors.textPrimary
+                    )
                 }
+            }
+            PrimaryText(
+                text = stringResource(R.string.no_collections_description2),
+                modifier = Modifier.fillMaxWidth(),
+                fontSize = 14.sp
+            )
+            if (normalTabsState.value!!.isNotEmpty()) {
+
+                Spacer(modifier = Modifier.height(12.dp))
+
+                NoCollectionsButtonItem(
+                    buttonText = "Save tabs to collection",
+                    onClick = onAddTabsToCollectionTapped
+                )
+            }
         }
     }
+}
 
-    companion object {
-        const val LAYOUT_ID = R.layout.no_collections_message
+/**
+ * No Collections Tabs Button
+ *
+ * @param buttonText The error button's text and accessibility hint.
+ * @param onClick The lambda called when the button is clicked.
+ */
+@Composable
+fun NoCollectionsButtonItem(buttonText: String, onClick: () -> Unit) {
```

Comment:
Is this Composable only used here? If so, the `buttonText` doesn't need to be a parameter